### PR TITLE
feat: added functions for insert/get from users table

### PR DESCRIPTION
**Pull request by ThomasLiARDJAVA**
# Administrative Info

## Changes
What changes did you make?
- Added 2 core model functions on the database end for signup/login: `getById` and `insert`

## Dependencies/Library installed: 
None

## Checklist

Remove any items which are not applicable.

- [x] I have performed a self-review of my own code
- [x] I have added tested my program thoroughly
- [x] Did you add/sub fields in the environment file? if so please update .env_sample

### CODE REVIEW

File: app/database/models/usersModel.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+/**
+ * insert user into users table
+ * @param user
+ * @returns {Promise<{status: string}>}
+ */
+async function insert(user){
+    try {
+        await db('users')
+            .insert(user);
+        return {status: 'success'};
```
Comment (rcherukuru2000) [inline app/database/models/usersModel.js:24]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/cse110-fa21-group17/cse110-fa21-group17/pullRequest?prid=8408698)
```suggestion
        return {status: "success"};
```

File: app/database/models/usersModel.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+}
+
+/**
+ * insert user into users table
+ * @param user
+ * @returns {Promise<{status: string}>}
+ */
+async function insert(user){
+    try {
+        await db('users')
```
Comment (rcherukuru2000) [inline app/database/models/usersModel.js:22]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/cse110-fa21-group17/cse110-fa21-group17/pullRequest?prid=8408698)
```suggestion
        await db("users")
```

File: app/database/models/usersModel.js [hunk 3]
```diff
@@ -0,0 +1,31 @@
+const db = require('../dbConfig');
+
+/**
+ * get user by id
+ * @param id
+ * @returns array []
+ */
+async function getById(id){
+    const rows = await db('users')
```
Comment (rcherukuru2000) [inline app/database/models/usersModel.js:9]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/cse110-fa21-group17/cse110-fa21-group17/pullRequest?prid=8408698)
```suggestion
    const rows = await db("users")
```

File: app/database/models/usersModel.js [hunk 4]
```diff
@@ -0,0 +1,31 @@
+const db = require('../dbConfig');
```
Comment (rcherukuru2000) [inline app/database/models/usersModel.js:1]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/cse110-fa21-group17/cse110-fa21-group17/pullRequest?prid=8408698)
```suggestion
const db = require("../dbConfig");
```

File: app/database/models/usersModel.js [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+
+/**
+ * get user by id
+ * @param id
+ * @returns array []
+ */
+async function getById(id){
+    const rows = await db('users')
+        .where({id})
+        .select('*');
```
Comment (rcherukuru2000) [inline app/database/models/usersModel.js:11]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/cse110-fa21-group17/cse110-fa21-group17/pullRequest?prid=8408698)
```suggestion
        .select("*");
```

File: app/database/models/usersModel.js [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+ * @returns {Promise<{status: string}>}
+ */
+async function insert(user){
+    try {
+        await db('users')
+            .insert(user);
+        return {status: 'success'};
+    } catch(err){
+        console.error(err);
+        return {status: 'failed'};
```
Comment (rcherukuru2000) [inline app/database/models/usersModel.js:27]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/cse110-fa21-group17/cse110-fa21-group17/pullRequest?prid=8408698)
```suggestion
        return {status: "failed"};
```

File: app/database/models/usersModel.js (hunk 4 above)
Comment (ThomasLiARDJAVA) [reply app/database/models/usersModel.js:1]: Not sure what Codacy is doing here... Maybe it's using a style guide I have not used before, feel like this should be a group discussion on which style guide to use. I was following JS style guide: 
https://google.github.io/styleguide/jsguide.html#features-string-literals

Comment (s8sun) [reply app/database/models/usersModel.js:1]: Don't worry about this too much now as we are setting up the CI/CD pipeline using git action. We are still testing things out. Ignore this auto generated style suggestions for now.

Comment (ThomasLiARDJAVA) [reply app/database/models/usersModel.js:1]: Do you guys still need the help? I just came back

**Review by rcherukuru2000 — APPROVED** [review]

**Pull request merged**